src/Raven.Server/Documents/Patch/BlittableObjectInstance.cs

```diff
@@ -142,36 +143,76 @@ private JsValue GetPropertyValue(string key, int propertyIndex)
+                        {
+                            item = TranslateToJs(null, null, blit.Item2, blit.Item1);
+                        }
+                        else
+                        {
+                            item = TranslateToJs(parent, null, blit.Item2, blit.Item1);
+                        }
+                        items[i] = new PropertyDescriptor(item, true, true, true);
+                    }
+                    jsArray = new ArrayInstance(e, items);
+                    jsArray.Prototype = e.Array.PrototypeObject;
+                    jsArray.Extensible = true;
+                }
+                else
+                {
+                    Dictionary<uint, PropertyDescriptor> items = new Dictionary<uint, PropertyDescriptor>(bjra.Length);
+                    for (uint i = 0; i < bjra.Length; i++)
+                    {
+                        var blit = bjra.GetValueTokenTupleByIndex((int)i);
+                        BlittableJsonToken itemType = blit.Item2 & BlittableJsonReaderBase.TypesMask;
+                        JsValue item;
+                        if (itemType == BlittableJsonToken.Integer || itemType == BlittableJsonToken.LazyNumber)
+                        {
+                            item = TranslateToJs(null, null, blit.Item2, blit.Item1);
+                        }
+                        else
+                        {
+                            item = TranslateToJs(parent, null, blit.Item2, blit.Item1);
+                        }
+                        items[i] = new PropertyDescriptor(item, true, true, true);
+                    }
+                    jsArray = new ArrayInstance(e, items);
+                    jsArray.Prototype = e.Array.PrototypeObject;
+                    jsArray.Extensible = true;
+                }
+                return jsArray;
+            }
 
             private unsafe JsValue TranslateToJs(BlittableObjectInstance owner, string key, BlittableJsonToken type, object value)
             {
+                
                 switch (type & BlittableJsonReaderBase.TypesMask)
                 {
                     case BlittableJsonToken.Null:
                         return JsValue.Null;
                     case BlittableJsonToken.Boolean:
-                        return (bool)value;
+                        return (bool)value?JsBoolean.True:JsBoolean.False;
                     case BlittableJsonToken.Integer:
-
                         // TODO: in the future, add [numeric type]TryFormat, when parsing numbers to strings
-                        owner.RecordNumericFieldType(key, BlittableJsonToken.Integer);                                                
+                        owner?.RecordNumericFieldType(key, BlittableJsonToken.Integer);                                                
 
-                        return (long)value;
+                        return new JsNumber((long)value);
```

Comment:
Casting to JsValue goes through conversion operator that has cache, allocation can be avoided for smaller numbers

Reply:
https://github.com/sebastienros/jint/blob/5d408955dc904300e543d27bb12ad10c19327b84/Jint/Native/JsValue.cs#L449